### DISCUSSION TITLE
硬连接和软连接的问题

### DISCUSSION THREAD

**Discussion by Detailed-x** [zhangyuanes/OperatingSystemExperiments#17] [Q&A]
假设文件F1的引用计数最初为1。首先，我们创建一个符号链接文件F2链接到F1，然后创建一个硬链接文件F3链接到F1。之后，F1被删除。现在，F2和F3的引用计数分别是。

我看到两个不同的答案：一个是1，1 一个是0，1
请问哪一个答案是对的？

**Comment by zhangyuanes** [comment]
> 假设文件F1的引用计数最初为1。首先，我们创建一个符号链接文件F2链接到F1，然后创建一个硬链接文件F3链接到F1。之后，F1被删除。现在，F2和F3的引用计数分别是。
> 
> 我看到两个不同的答案：一个是1，1 一个是0，1
> 请问哪一个答案是对的？

正确答案： 1， 1
- 软链接：新建立的一个文件，这个文件内容就是专门用来指向别的文件的，此题中F2中存放F1的inode，并不会影响原有文件的count使用计数，软链接count总是为1；
- 硬链接：硬链接与原文件是等价的，Innode值是相同的，count值等于硬链接的个数，原文件与硬链接互为硬链接，删除硬连接或者源文件，都不会真实删除，而是count-1，直到count=0的时候才会真正删除。此题中F3链接到F1，F1的count+1 = 2，F1删除后count-1 = 1。
